core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -1864,6 +1780,18 @@ class ConnectionQuotas(config: KafkaConfig, time: Time, metrics: Metrics) extend
       sensor
     }
   }
+
+  /**
+   * Close `channel` and decrement the connection count.
+   */
+  def closeChannel(listenerName: ListenerName, channel: SocketChannel): Unit = {
+    if (channel != null) {
+      debug(s"Closing connection from ${channel.socket.getRemoteSocketAddress}")
+      dec(listenerName, channel.socket.getInetAddress)
+      closeSocket(channel, this)
```

Comment:
Well, the ConnectionQuotas class is tracking how many connections exist. It needs to be informed when a connection is closed. If you want, I can have the function take a Logging parameter so that we can log until `SocketServer` (or whatever) rather than `SocketServer.ConnectionQuotas`. But this is a DEBUG level message anyway... realistically, you don't ever see this in prod, so... not sure if it matters.

Reply:
I guess to me, when you have something that is limiting the number of X objects, it makes sense for that thing to be logging about opening / closing the X objects as well. Then you can add helpful info about how many X objects there are, and so forth (in theory, I don't think we do that here).

So really this thing shouldn't be a "quotas" class but more like a pool or something. But we've changed enough in this PR, so let's not go there :)

Anyway, I will move it around, although we'll end up with some 4 or 5 argument utility functions.

Reply:
Update: I have changed it so that `closeChannel` takes a `Logging` argument (which will be either the Acceptor or Processor). This should lead to the log message about closure being listened under either the Acceptor or Processor, as it was before this PR.